File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        int colonIndex = logLine.indexOf(":");
        String message = logLine.substring(colonIndex + 1);
        return message.trim();
    }

    public static String logLevel(String logLine) {
        int start = logLine.indexOf("[") + 1;
        int ende = logLine.indexOf("]");
        String level = logLine.substring(start, ende);
        return level.toLowerCase();
    }

    public static String reformat(String logLine) {
        String letzterTeil = message(logLine);
        String ersterTeil = logLevel(logLine);
        String Ergebnis = letzterTeil + " (" + ersterTeil + ")";
        return Ergebnis;
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
    int pos = line.find(":");
    return line.substr(pos + 2);
}

std::string log_level(std::string line) {
    int anfang = line.find("[");
    int ende = line.find("]");
    return line.substr(anfang+1, ende-1);
}

std::string reformat(std::string line) {
    std::string msg = message(line);
    std::string lvl = log_level(line);
    return msg + " " + "(" + lvl + ")";
}
}  // namespace log_line
